### PR TITLE
Fix ruby dependent extensions.

### PR DESCRIPTION
**Pull request by Kerilk**
Hello,

While trying to package ruby extensions for spack, I encountered several dependency issues between gems. They are related to how `GEM_PATH` is handled by spack and by Ruby `gem`.
 - Arguably `GEM_PATH` and `PYTHONPATH` should be treated similarly, so I mimicked how python is dealing with `PYTHONPATH`
 - Not using the `--install-dir` option of `gem` (relying on `GEM_HOME` to set it), causes `GEM_PATH` to be removed from the environment, so I added the `--install-dir` option to `gem install`. See below link for the problematic line of code.

https://github.com/rubygems/rubygems/blob/4ebde70687e5ec158537e852a34ac3cae1674647/lib/rubygems/commands/install_command.rb#L160

This fixes my issues and allows gems dependencies to be resolved at build time. I am very unfamiliar with spack, so I may have ended far the correct solution though.

Thanks,
Brice

### CODE REVIEW

**Review by alalazo — DISMISSED** [review]
Hi Brice, thanks for the contribution! I'll wait for @adamjstewart review on this one, but ping me in a few days if you don't see the discussion moving forward.

**Comment by Kerilk** [comment]
Hi Massimiliano, it's been quite some time! Thanks for the review.

**Review by adamjstewart — DISMISSED** [review]
Thanks @Kerilk, this looks great! I don't think there are many Spack developers who are familiar with Ruby. Would you have any interest in helping to maintain our Ruby packages? You don't have to be a Spack expert (@alalazo and I can handle the Spack incatricies) but it would be great to have someone who can answer Ruby questions. You can either add `maintainers = ['Kerilk']` to the `ruby` package if you only want to maintain Ruby or add it to `ruby` and the `RubyPackage` build system if you want to maintain all Ruby packages (currently we only have 17).

**Review by adamjstewart — APPROVED** [review]

**Comment by Kerilk** [comment]
@adamjstewart Thanks for the review. I've added myself as maintainer for both as I'll most likely package several gems in the near future. Unfortunately I am not sure I am the maintainer Ruby deserves...

**Comment by adamjstewart** [comment]
Thanks @Kerilk! The `maintainers` variable is a list, so if anyone else comes along who knows Ruby better, they can also add themselves as maintainers. 1 maintainer is much better than 0 maintainers!

**Review by alalazo — APPROVED** [review]

**Pull request merged**
